web/handler/pizzas.ml

```diff
@@ -1,8 +1,72 @@
 let index req =
   let open Lwt.Syntax in
+  let csrf = Sihl.Web.Csrf.find req |> Option.get in
+  let alert = Sihl.Web.Flash.find_alert req in
+  let notice = Sihl.Web.Flash.find_notice req in
   let* user = Service.User.Web.user_from_session req |> Lwt.map Option.get in
-  Lwt.return @@ Sihl.Web.Response.of_html (View.Pizzas.index user)
+  let* pizzas = Pizza.find_pizzas () in
+  let* ingredients = Pizza.find_ingredients () in
+  Lwt.return
+  @@ Sihl.Web.Response.of_html
+       (View.Pizzas.index user ~alert ~notice csrf pizzas ingredients)
 ;;
 
-let create _ = failwith "todo pizza create"
-let delete _ = failwith "todo pizza delete"
+let show req =
+  let open Lwt.Syntax in
+  let alert = Sihl.Web.Flash.find_alert req in
+  let notice = Sihl.Web.Flash.find_notice req in
+  let* user = Service.User.Web.user_from_session req |> Lwt.map Option.get in
+  let name = Sihl.Web.Router.param req "name" in
+  let* pizza = Pizza.find_pizza name in
+  match pizza with
+  | None ->
+    Sihl.Web.Response.redirect_to "/pizzas"
+    |> Sihl.Web.Flash.set_alert (Format.sprintf "Pizza '%s' not found" name)
+    |> Lwt.return
+  | Some pizza ->
+    Lwt.return
+    @@ Sihl.Web.Response.of_html (View.Pizzas.show user ~alert ~notice pizza)
+;;
+
+let create req =
+  let open Lwt.Syntax in
+  let* name = Sihl.Web.Request.urlencoded "name" req in
+  match name with
+  | None ->
+    Sihl.Web.Response.redirect_to "/pizzas"
+    |> Sihl.Web.Flash.set_alert "Invalid input provided"
+    |> Lwt.return
+  | Some name ->
+    let* ingredients = Sihl.Web.Request.urlencoded_list "ingredients" req in
+    if List.length ingredients < 1
```

Comment:
Your error case when there are no ingredients is very similar to the error case when no name is provided. The only difference is their error messages. Could you try to do the validation of both the name and the ingredients in the beginning of the handler and then react accordingly? One approach could be that your validation step returns `None` if everything was alright and `Some msg` with the error message. You could pattern match on this and in the `None` branch handle the happy path (pizza creation, etc.) and in the `Some msg` branch you do the `set_alert` stuff.